superset-frontend/src/components/Select/Select.tsx

```diff
@@ -63,12 +62,7 @@ type PickedSelectProps = Pick<
   | 'value'
 >;
 
-type OptionsProps = Exclude<AntdSelectAllProps['options'], undefined>;
-
-export interface OptionsType extends Omit<OptionsProps, 'label'> {
-  label?: string;
-  customLabel?: ReactNode;
-}
```

Comment:
@ktmud The objective of the extension is to change the type of `label` from `React.Node` to `string` and to introduce another property called `customLabel` that accepts a `React.Node`. You can search from `customLabel` to see where it's being used.

Reply:
Then it should extend `OptionData`, not `OptionsProp`.

Reply:
I'm also curious why did we decide to introduce a `customLabel` prop instead of using [optionRenderer](https://github.com/apache/superset/blob/46715b295c3026c3eb1ee439f1d16fc776e61372/superset-frontend/src/components/Select/DeprecatedSelect.tsx#L87) like in the old Select.

Reply:
I can't remember any particular reason. I guess it was just a style choice when we were designing the API. I don't see any problem in changing that to `optionRenderer` if you want. It's being used only by 10 files so it would be really easy to migrate.

@geido Please correct me if I'm forgetting anything.

Reply:
@ktmud Just let me know if you're going to make the change or add the `customLabel` back to the type so I can continue the review.

Reply:
@michael-s-molina I don't think we should add it back since it's incorrect code.

Reply:
What I mean is to restore the original intention of the label and customLabel under the correct extended type `OptionData` or introduce the concept of `optionRenderer`. If we merge as it is we don't have any type checking regarding custom labels.

Reply:
The existing check doesn't work already. I was thinking of some bigger refactoring with the customLabel check so I'd rather keep it in a separate PR. I can put that PR up first if you prefer.

Reply:
No problem making the change in a follow-up PR. I'll just do a final round of testing tomorrow morning but if I don't find anything we can merge this PR 👍